addons/web/static/src/core/l10n/dates.js

```diff
@@ -615,3 +615,150 @@ export function deserializeDateTime(value, options = {}) {
             numberingSystem: Settings.defaultNumberingSystem,
         });
 }
+
+/**
+ * Tries to parse a datetime object from a time string
+ * representation such as:
+ * "10:15"  -> 10:15:00
+ * "2h5"    -> 02:50:00
+ * "1015"   -> 10:15:00
+ * "125"    -> 12:50:00
+ * "315"    -> 03:15:00
+ * "5:15pm" -> 17:15:00
+ *
+ * Returns null if the value could not be parsed.
+ *
+ * @param {string} value
+ * @returns {DateTime}
+ */
+export function parseTime(value, parseSeconds) {
+    const { isPm, isAm } = meridiemCheck(value);
+    value = normalizeTime(value);
+
+    if (!value.match(/[0123456789:]+/)) {
```

Comment:
`RegExp.prototype.test` is a better approach when not using the returned match value
```suggestion
    if (!/[\d:]+/.test(value)) {
```